analyzers/src/SonarAnalyzer.CSharp/Rules/DatabasePasswordsShouldBeSecure.cs

```diff
@@ -130,21 +131,25 @@ private void ReportEmptyPassword(XDocument doc, string webConfigPath, Compilatio
             }
         }
 
-        private void ReportEmptyPassword(JObject doc, string appSettingsPath, CompilationAnalysisContext c)
+        private void ReportEmptyPassword(JsonNode doc, string appSettingsPath, CompilationAnalysisContext c)
         {
-            foreach (var connectionStringToken in doc.SelectToken("ConnectionStrings").Children())
+            if (doc.ContainsKey("ConnectionStrings"))
             {
-                if (connectionStringToken is JProperty connectionStringProperty && IsVulnerable(connectionStringProperty.Value.ToString()))
+                foreach (var key in doc["ConnectionStrings"].Keys)
                 {
-                    c.ReportIssue(Diagnostic.Create(Rule, CreateLocation(connectionStringProperty.Value, connectionStringProperty.Value.ToString().Length, appSettingsPath)));
+                    if (doc["ConnectionStrings"][key] is var connectionStringNode && IsVulnerable(connectionStringNode.Value.ToString()))
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, CreateLocation(connectionStringNode, appSettingsPath)));
+                    }
                 }
             }
         }
 
-        private static Location CreateLocation(IJsonLineInfo lineInfo, int length, string path)
+        private static Location CreateLocation(JsonNode node, string path)
         {
-            var start = new LinePosition(lineInfo.LineNumber - 1, lineInfo.LinePosition - 1 - length);
-            var end = new LinePosition(lineInfo.LineNumber - 1, lineInfo.LinePosition - 1);
+            var length = node.Value.ToString().Length;
```

Comment:
Is this correct for things like:

``` JSON
{
   value: 123.000,
}
```
Or is the value still a JsonNode here? (and not a floating-point or decimal)

Reply:
This method is used on the strings that passed the check inside _IsVulnerable_, so they are already supposed to be strings